bluemix/src/main/java/com/ibm/watson/SpeechToText.java

```diff
@@ -43,120 +39,157 @@ public String speechToText(byte[] incomingBinaryPOSTData) {
-        }finally {
+            return failXML(e, "Bluemix ");
+        } finally {
             try {
                 if (conn != null) {
                     conn.disconnect();
                 }
             } catch (Exception e) {
                 e.printStackTrace();
-                return XML_DEC + "<results>" + STATUS_ERROR + "<statusInfo>Bluemix " + e.getClass().getName() + ": " + e.getMessage() + "</statusInfo><transcript /></results>";
+                return failXML(e, "Bluemix ");
             }
         }
-        
+
+    }
+
+
+    // Set up a connection with the speech-to-text service
+    private HttpURLConnection setupHttpConnection(byte[] incomingBinaryPOSTData,
+                                                  String serviceURL,
+                                                  JsonObject credentials) throws IOException {
+
+        String serviceUser = credentials.getString("username");
+        String servicePass = credentials.getString("password");
+        String auth = serviceUser + ":" + servicePass;
+
+        // Prepare the query string parameters.
+        // See the Speech to Text service documentation for details.
+        String queryString = "&model=en-US_NarrowbandModel"
+                + "&continuous=true";
+        if (DEBUG) {
+            System.out.println("POSTing [" + incomingBinaryPOSTData.length + "] bytes");
+            System.out.println("Using URL [" + serviceURL + API_PATH + "?" + queryString + "]");
+        }
+
+        // Prepare the HTTP connection to the service
+        HttpURLConnection conn = (HttpURLConnection) new URL(serviceURL + API_PATH + "?" + queryString).openConnection();
+
+        // Set up our header with all the information we need to make the api request
+        conn.setDoInput(true);
+        conn.setDoOutput(true);
+        conn.setRequestMethod("POST");
+        conn.setRequestProperty("Authorization", "Basic " + Base64.encodeBase64String(auth.getBytes()));
+        conn.setRequestProperty("Accept", "*/*");
+        conn.setRequestProperty("Content-Type", "audio/flac");
+        conn.setRequestProperty("Transfer-encoding", "chunked");
+        conn.setFixedLengthStreamingMode(incomingBinaryPOSTData.length);
+
+        // Uncomment the following to opt out of improving the service with this audio file
+        //	    conn.setRequestProperty("X-WDC-PL-OPT-OUT", "1");
+        return conn;
+    }
+
+
+    private void sendRequest(byte[] incomingBinaryPOSTData, HttpURLConnection conn) throws IOException {
+        OutputStream output = conn.getOutputStream();
+        output.write(incomingBinaryPOSTData);
+        output.flush();
+        output.close();
+    }
+
+
+    // Collect the transcripts from all the results in our response body
+    private String readTranscripts(JsonObject jsonResponse) {
+        JsonArray resultsArray = jsonResponse.getJsonArray("results");
```

Comment:
This jsonResponse.getJsonArray("results") used to be inside the try/catch, presumably in the case where there was no array named "results" in the response, so that we can properly blame that on the service.